react/src/lib/components/DeckGLMap/components/DistanceScale.tsx

```diff
@@ -1,26 +1,52 @@
 import React from "react";
 
 interface scaleProps {
+    // Needed the zoom value to calculate width in units
     zoom: number;
+    // Scale increment value
+    incrementValue: number;
+    // width per unit value to calculate width in units
+    widthPerUnit: number;
 }
-const DistanceScale: React.FC<scaleProps> = ({ zoom }: scaleProps) => {
-    const [width] = React.useState(100);
-    const pixelPerUnit = width / Math.pow(2, zoom);
-    const scaleBarStyle = {
-        width: width,
+
+const roundOffvalues = function (num: number, step: number) {
+    return Math.floor(num / step + 0.5) * step;
+};
+
+const DistanceScale: React.FC<scaleProps> = ({
+    zoom,
+    incrementValue,
+    widthPerUnit,
+}: scaleProps) => {
+    const [rulerWidth, setRulerWidth] = React.useState<number>(widthPerUnit);
+    const widthInUnits = widthPerUnit / Math.pow(2, zoom);
+    const scaleRulerStyle = {
+        width: rulerWidth,
         height: "4px",
         border: "2px solid gray",
         borderTop: "none",
         display: "inline-block",
         marginLeft: "3px",
+        marginRight: "3px",
         right: 0,
         bottom: 0,
     };
 
+    let scaleValue = 0;
+
+    scaleValue =
+        widthInUnits < incrementValue
+            ? Math.round(widthInUnits)
+            : roundOffvalues(widthInUnits, incrementValue);
```

Comment:
`scaleValue` can be `const`. Also it's redundant to initialize it to 0 and then immediately set it to another value. You can do it in one step:
```js
     const scaleValue = widthInUnits < incrementValue
            ? Math.round(widthInUnits)
            : roundOffvalues(widthInUnits, incrementValue);
```